src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/TokenCertificatesApiController.java

```diff
@@ -203,14 +203,31 @@ public TokenCertificatesApiController(TokenCertificateService tokenCertificateSe
 
     @Override
     @PreAuthorize("hasAuthority('SEND_AUTH_CERT_DEL_REQ')")
-    public ResponseEntity<Void> unregisterCertificate(String hash) {
+    public ResponseEntity<Void> unregisterAuthCertificate(String hash) {
         try {
             tokenCertificateService.unregisterAuthCert(hash);
         } catch (CertificateNotFoundException e) {
             throw new ResourceNotFoundException(e);
         } catch (GlobalConfService.GlobalConfOutdatedException | TokenCertificateService.InvalidCertificateException
                 | TokenCertificateService.SignCertificateNotSupportedException e) {
             throw new BadRequestException(e);
+        } catch (ActionNotPossibleException | KeyNotFoundException
+                | TokenCertificateService.NoValidAuthCertificateException e) {
+            throw new ConflictException(e);
+        }
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('SEND_AUTH_CERT_DEL_REQ')")
+    public ResponseEntity<Void> markAuthCertForDeletion(String hash) {
+        try {
+            tokenCertificateService.markAuthCertForDeletion(hash);
+        } catch (CertificateNotFoundException e) {
+            throw new ResourceNotFoundException(e);
+        } catch (GlobalConfService.GlobalConfOutdatedException | TokenCertificateService.InvalidCertificateException
```

Comment:
Not a change request about this PR, but a thought. Are we misusing http 400 bad request? Like "global conf outdated" is not a client error, it is not result of a bad API call? Perhaps some other catches here as well.

But it is a widely used pattern in our implementation, so not directly related to this PR.

Reply:
That is a good point. I think that 400 Bad Request is being misused pretty widely in the Internet and it is sometimes considered a general exception even though it should be on the client.